### DISCUSSION TITLE
Is there any option to hoist the `node_modules/.bin` directory of a submodule to root?

### DISCUSSION THREAD

**Discussion by j-murata** [pnpm/pnpm#6837] [Q&A]
My monorepo project has an npm script that clears `node_modules` in the submodule `package.json` as follows:
```console
(monorepo-root)
├── packages
│   ├── submodule-a
│
├── pnpm-workspace.yaml
```
```json
{
  "name": "@monorepo/submodule-a",
  "devDependencies": {
    "rimraf": "*",
    ...
  },
  "scripts": {
    "clear": "rimraf ./node_modules"
  }
}
```
[`rimraf`]: https://www.npmjs.com/package/rimraf
When it runs on submodule, it cannot remove itself if it refers to [`rimraf`] in the submodule `node_modules/.bin`, so I want to set it up so that [`rimraf`] installed in root `.bin` is used.

I know that setting [`node-linker` to `hoisted`](https://pnpm.io/npmrc#node-linker) will achieve this, as it will install only on root `node_modules`, but I don't want to use `node-linker=hoisted` if possible, and am looking for another way to achieve this.

I was able to set [`dedupe-direct-deps=true`](https://pnpm.io/npmrc#dedupe-direct-deps) to prevent the [`rimraf`] body js files from being created under submodule `node_modules/`, but `node_modules/.bin` is still created and this PATH is referenced.

If anyone has a better method, I would appreciate it if you could let me know.

**Comment by Jason3S** [comment]
@j-murata,

I don't think it is necessary to specify `rimraf` in the submodule `devDependencies`. It should work fine only having dev tools in the repo root.

**Comment by j-murata** [comment]
@Jason3S Thanks for your comment.

I think you are right, the script will work even without installing `rimraf` in subprojects.
However, I don't think it is good for the portability of individual subprojects that npm script depends on modules that are not installed (i.e., it depends on belonging to monorepo).
